app/scripts/proactive/model/GroupByProjectMixin.js

```diff
@@ -10,13 +10,83 @@ define(
 
         var defaultProjectName = 'Default';
 
-        function sortDefaultFirst(keys) {
+        function sortDefaultFirst(groups, option) {
+            var keys = _.keys(groups);
+            var values = _.values(groups);
+            if (option === undefined || option == "A-Z") {
+                keys = sortProjectsAsc(keys);
+            } else if (option == "Z-A") {
+                keys = sortProjectsDesc(keys);
+            } else if (option == "Newer edit") {
+                keys = sortProjectsByNewerEdit(keys, values);
+            } else if (option == "Older edit") {
+                keys = sortProjectsByOlderEdit(keys, values);
+            }
+            return keys;
+        }
+
+       function sortProjectsAsc(keys) {
             keys.sort();
             if (_.indexOf(keys, defaultProjectName) != -1) {
                 keys = _.without(keys, defaultProjectName);
                 keys.unshift(defaultProjectName);
             }
             return keys;
+       }
+
+       function sortProjectsDesc(keys) {
+            keys.sort().reverse();
+            if (_.indexOf(keys, defaultProjectName) != -1) {
+                keys = _.without(keys, defaultProjectName);
+                keys.push(defaultProjectName);
+            }
+            return keys;
+       }
+
+       function sortProjectsByNewerEdit(keys, values) {
+            keys.sort(function(model1, model2) {
+                var index1 = getIndexOfWorkflow(keys, model1);
+                var index2 = getIndexOfWorkflow(keys, model2);
+                var model1Date = getModifyDateOfProject(values, index1, "Newer edit");
+                var model2Date = getModifyDateOfProject(values, index2, "Newer edit");
+                return model2Date - model1Date;
+            });
+            return keys;
+       }
+
+       function sortProjectsByOlderEdit(keys, values) {
+            keys.sort(function(model1, model2) {
+                var index1 = getIndexOfWorkflow(keys, model1);
+                var index2 = getIndexOfWorkflow(keys, model2);
+                var model1Date = getModifyDateOfProject(values, index1, "Older edit");
+                var model2Date = getModifyDateOfProject(values, index2, "Older edit");
+                return model1Date - model2Date;
+            });
+            return keys;
+       }
+
+       function getIndexOfWorkflow(keys, key) {
+            var index =0;
+            for (var i = 1; i < keys.length; i ++) {
+                if (keys.at(i) === key) {
```

Comment:
Using keys.at(i) works, but keys[i] is more commonly used for arrays and slightly more efficient in this context.